devlib/target.py

```diff
@@ -701,6 +703,33 @@ def _install_module(self, mod, **params):
     def _resolve_paths(self):
         raise NotImplementedError()
 
+    def is_network_connected(self):
+        self.logger.debug('Checking for internet connectivity...')
+        return self._ping_server(GOOGLE_DNS_SERVER_ADDRESS)
+
+    def _ping_server(self, ip_address, timeout=1, packet_count=1):
+        command = 'ping -q -c {} -w {} {}'.format(packet_count, timeout, ip_address)
+
+        # Ideally we'd like to just run with check_exit_code=False, then check
```

Comment:
Not entirely sure about how it plays with self.execute, but... what about just using something like:

```python
command = 'ping -q -c {} -w {} {}; echo $?'.format(packet_count, timeout, ip_address)
output = self.execute(command)
if output == '0':
    # Success
else
   # Failure
```

Reply:
`execute` takes care of that. All you need to do is

```
try:
     self.execute(command)
     return True
except TargetError:
     return False
```